src/libs/antares/study/parts/short-term-storage/properties.cpp

```diff
@@ -70,25 +70,40 @@ unsigned int groupIndex(Group group)
     }
 }
 
-bool Properties::loadKey(const IniFile::Property* p)
+template<class T>
+static bool valueForOptional(const IniFile::Property* property, std::optional<T>& out)
 {
-    auto valueForOptional = [p](std::optional<double>& opt) {
-        if (double tmp; p->value.to<double>(tmp))
-        {
-            opt = tmp;
-            return true;
-        }
+    if (double tmp; property->value.to<double>(tmp))
+    {
+        out = tmp;
+        return true;
+    }
+    return false;
+}
+
+template<class T>
+static bool checkMandatory(const std::string& name,
+                           const std::optional<T>& property,
+                           const std::string& label)
+{
+    if (!property.has_value())
+    {
+        logs.error() << "Property " << label << " is mandatory for short term storage " << name;
         return false;
-    };
+    }
+    return true;
+}
 
+bool Properties::loadKey(const IniFile::Property* p)
+{
     if (p->key == "injectionnominalcapacity")
-        return valueForOptional(this->injectionCapacity);
+        return valueForOptional(p, this->injectionCapacity);
```

Comment:
Take the example of **injectionCapacity** : 
At some point we have : 
```C++
if (p->key == "injectionnominalcapacity")
    return valueForOptional(p, this->injectionCapacity);
```
and then : 
```C+
if (!checkMandatory(name, injectionCapacity, "injectionnominalcapacity"))
    return false;
```
**Question** : isn't that too complicated ?
Couldn't we have : 
```C++
if (p->key == "injectionnominalcapacity")
    return p->value.to<std::optional<double>>(this->injectionCapacity);
...
if (!injectionCapacity.has_value())
{
    // Some logs here
    return false;
}
```
? 
In other words, **if I'm not missing something**, we could remove functions **valueForOptional(...)** and **checkMandatory(...)**

Reply:
`return p->value.to<std::optional<double>>(this->injectionCapacity);

Applied this, should we simply log instead of using checkMandatory() ? @flomnes

Reply:
@payetvin : you told me that 
```C++
if (p->key == "injectionnominalcapacity")
    return p->value.to<std::optional<double>>(this->injectionCapacity);
...
```
make unit tests fail at execution.
Could you try this : 
```C++
if (p->key == "injectionnominalcapacity")
    return p->value.to<double>(*this->injectionCapacity);
...
```

Reply:
Same results, injectionCapacity.has_value() is false

Reply:
ok, the unit tests fail because : 
- when using `p->value.to<std::optional<double>>(this->injectionCapacity);` : there is no template specialization of the `to<T>` for **std::optional<double>**, so a default specialization is used, which does nothing.
- when using `p->value.to<double>(*this->injectionCapacity);`, the internal value of the **std::optional** is changed correctly, but not the internal status bool **_has_value_** which remains to **false**

I tried find a quick and clean solution to this, in file **src/ext/yuni/src/yuni/core/string/traits/into.h**, but I can't see any.
So I give up.

Reply:
Eventually this problem was solved.
@flomnes : thx for your precious technical help !